### PR TITLE
Add /leaderboard slash command

### PR DESCRIPTION
**Pull request by Madjosz**
* `getLeaderboard` to access Codewars API endpoint
* extract `languageAutocomplete`, `findLanguage`, `checkBotPlayground` from `rankup.ts` to `common.ts`
* add dependency to `table-6.8.0` https://www.npmjs.com/package/table
* unify error handling

Overall Leaderboard

### CODE REVIEW

File: src/commands/leaderboard.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+function format(
+  lang: Language | null,
+  leaderboard: LeaderboardPosition[]
+) {
+  const data = [['#','username','score','rank'], ...leaderboard.map(u=>[u.position, u.username, u.score, getRank(u.rank)])];
+  return `:trophy: **${lang?.name ?? "Overall"} leaderboard** :medal:\n`+"```\n"+table(data, tableConfig)+"\n```";
+}
+function getRank(rank: number) {
+  if(rank > 0) return rank + " dan";
```
Comment (kazk) [inline src/commands/leaderboard.ts:54]: Please format the code with Prettier. https://github.com/codewars/discord-bot#code-style

How did you make this PR? Running `npm install` should install the `pre-commit` hook to autoformat the changed code.

File: src/commands/leaderboard.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      option
+        .setName("language")
+        .setDescription("The programming language to show leaderboard for")
+        .setAutocomplete(true)
+    )
+    .addIntegerOption((option) =>
+      option.setName("startposition")
+        .setDescription("The top shown position (default 1)")
+        // limits seem not to be checked
+        .setMinValue(1)
```
Comment (kazk) [inline src/commands/leaderboard.ts:33]: Can you clarify this comment?

**Review by kazk — CHANGES_REQUESTED** [review]
Looks good to me, but please format the code. I also asked a question about a comment.

Next time, please consider refactoring in a separate commit, so it's easier to review the changes.

File: src/commands/leaderboard.ts (hunk 2 above)
Comment (Madjosz) [reply src/commands/leaderboard.ts:33]: This setting has no effect. I am still able to input numbers out of that range so there is no enforcement of this limits on Discord side.

File: src/commands/leaderboard.ts (hunk 1 above)
Comment (Madjosz) [reply src/commands/leaderboard.ts]: I had to run `npx pretty-quick` manually in the end.

**Comment by Madjosz** [comment]
I have splitted refactoring and new feature into two separate commits now and ran `prettier` on the files.

**Review by kazk — APPROVED** [review]
Thanks!

**Comment by kazk** [comment]
Deploy failed :(
I'll look into it.

**Comment by kazk** [comment]
Deployed :tada:

**Pull request merged**